performance_test/CMakeLists.txt

```diff
@@ -38,11 +38,10 @@ endif()
 # Use ODB support to create SQL database
 option(PERFORMANCE_TEST_ODB_FOR_SQL_ENABLED "Enable ODB for SQL" OFF)
 
-# Build code using ROS Eloquent packages
-option(ROS_ELOQUENT_ENABLED "Use ROS Eloquent" OFF)
-
-if(ROS_ELOQUENT_ENABLED)
-  add_definitions(-DROS_ELOQUENT)
+if(DEFINED ENV{ROS_DISTRO})
```

Comment:
This will stop working for ROS 2 F, is there any other way to detect the version of ROS that's being used? I've looked into the header files in case there's a constant, but I can't seem to find anything.

Reply:
@daggarwa @bmarchi I've looked more into this, and I can't find any way other than using the `ROS_DISTRO` variable. We can do the following to make this logic a bit more robust:

- Check the `ROS_VERSION` environment variable to make sure we're using ROS 2
- Add the `-DROS_ELOQUENT` definition if the `ROS_DISTRO` environment variable starts with a letter greater or equal to `eloquent`

I'd also suggest two more changes:

- Invert the logic, such that the new API is used, unless a definition is given (e.g. `USE_LEGACY_QOS_API`), that way we can remove that code in the future more easily
- Check that the `ROS_DISTRO` variable starts with a letter lower or equal to `d` (i.e. `dashing`) and then pass `USE_LEGACY_QOS_API`

Reply:
@esteve , what do you think of looking for the rclcpp major/minor version instead of seeing the letter? Or any package that we can use given a version. Also, why not using just one env variable rather than 2? I'm fine with any, just asking

Reply:
@BMarchi where can you query the major/minor version of rclcpp? I had a look at the header files and couldn't find anything.

The reason for using the two variables is because checking ROS_DISTRO does not tell us that we're using ROS 2, just the name of the distro.

Reply:
I pushed a commit of how we can use the rclcpp package version. Tell me what do you think about it.

Reply:
@BMarchi thanks. Looks good to me, what do you think of reversing the logic so that for anything lower than 0.7.0, CMake passes `USE_LEGAGY_QOS_API` instead?

Reply:
> @BMarchi thanks. Looks good to me, what do you think of reversing the logic so that for anything lower than 0.7.0, CMake passes `USE_LEGAGY_QOS_API` instead?

@BMarchi any updates on this?

Reply:
Sorry, I was busy yesterday. I pushed the new change

Reply:
@BMarchi I tried to compile the performance test with `USE_LEGAGY_QOS_API` flag enabled against the FastRTPS  version we have in ApexOS. But the compilation fails for me: 
```bash
divya.aggarwal@ade:~/perf_test_ws$ source ~/gc/apex_ws/install/setup.bash 
not found: "/home/divya.aggarwal/perf_test_ws/install/local_setup.bash"
divya.aggarwal@ade:~/perf_test_ws$ colcon build --cmake-clean-cache --cmake-args -DCMAKE_BUILD_TYPE=Release -DPERFORMANCE_TEST_FASTRTPS_ENABLED=ON -USE_LEGACY_QOS_API=ON
```